app/electron/main.ts

```diff
@@ -18,13 +18,42 @@ const defaultPort = 4466;
 
 function startServer(flags: string[] = []): ChildProcessWithoutNullStreams {
   const serverFilePath = path.join(process.resourcesPath, './electron/server');
-  return spawn(serverFilePath, [...flags], { shell: true });
+
+  const options = {shell: true, detached: false};
+  if (process.platform !== 'win32') {
+    // This makes the child processes a separate group, for easier killing.
+    options.detached = true;
+  }
+
+  return spawn(serverFilePath, [...flags], options);
 }
 
 let serverProcess: ChildProcessWithoutNullStreams | null;
 let intentionalQuit: boolean;
 let serverProcessQuit: boolean;
 
+function quitServerProcess() {
+  if (serverProcess) {
+    if (serverProcessQuit) {
+      log.error('server process already not running');
+    } else {
+      intentionalQuit = true;
+      log.info('stopping server process...');
+      if (process.platform !== 'win32') {
+        // This should kill the whole group of processes.
+        process.kill(-serverProcess.pid);
```

Comment:
Why the `-`?

Reply:
This should kill the whole group of processes (including the shell). See https://azimi.me/2014/12/31/kill-child_process-node-js.html

Reply:
Updated the comment.